From 6872a2e442ca27dd76d82ce6dd0f1fc5f499a71b Mon Sep 17 00:00:00 2001
From: Vignesh Sridharan <vignsrid@qti.qualcomm.com>
Date: Mon, 13 Jul 2026 12:04:30 +0530
Subject: [PATCH] bpl: Fix missing SMD param cleanup

SMD prams were not cleaned up when smd policy is not part of BSS config, causing
wireless file to contain stale SMD param entires even if they were removed from hapd
configs. This commit fixes it by cleanup stale SMD params when SMD policy is not
part of BSS config.
---
 framework/platform/bpl/cfg/uci/bpl_cfg_wifi.cpp | 10 ++++++++++
 1 file changed, 10 insertions(+)

diff --git a/framework/platform/bpl/cfg/uci/bpl_cfg_wifi.cpp b/framework/platform/bpl/cfg/uci/bpl_cfg_wifi.cpp
index 2def3bc76..f94b5e670 100644
--- a/framework/platform/bpl/cfg/uci/bpl_cfg_wifi.cpp
+++ b/framework/platform/bpl/cfg/uci/bpl_cfg_wifi.cpp
@@ -1031,6 +1031,16 @@ bool bpl_cfg_set_wifi_credentials(const std::string &iface,
                 options["smd_type"]            = std::to_string(config.smd_type);
                 options["smd_ptk_mode"]        = std::to_string(config.smd_ptk_mode);
                 options["smd_dl_drain_time"]   = std::to_string(config.smd_default_dl_drain_time);
+            } else {
+                // Remove SMD policy params if previously set in the wireless UCI file
+                options["smd_ap"]              = "";
+                options["smd_identifier"]      = "";
+                options["smd_timeout"]         = "";
+                options["smd_dl_data_fwd"]     = "";
+                options["smd_max_peer_apmlds"] = "";
+                options["smd_type"]            = "";
+                options["smd_ptk_mode"]        = "";
+                options["smd_dl_drain_time"]   = "";
             }
 
             return options;
-- 
2.34.1

